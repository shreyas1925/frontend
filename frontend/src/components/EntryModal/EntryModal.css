.modal {
    /* background-color: rgba(0, 0, 0, 0.5); */
    display: flex;
    justify-content: center;
    align-items: center;
    position: fixed;
}

.modal-content {
    margin-left: 55%;
    height: inherit;
    border-radius: 20px;
    padding: 30px;
    width: 1000px;
    margin-top: -70px;
    background-color: white;
}

.modal-header,
.modal-footer,
.modal-body {
    padding: 10px;

}

.modal-header {
    color: rgba(89, 89, 89, 255);
    font-weight: 700;
    font-size: 20px;
    padding: 70px 0 50px 0;
    font-weight: bold;
    margin-top: -35px;
}

.modal-body {
    height: 70vh;
    font-weight: 700;
    color: rgba(89, 89, 89, 255);
    font-size: 13px;
    display: flex;
    flex-direction: column;
}

.modal-body input {
    margin-top: 10px;
    width: 100%;
    height: 35px;
    border-radius: 5px;
    border: none;
    box-shadow: 1px 1px 1px 1px rgba(0, 0, 0, 0.5);
    margin-bottom: 20px;
    padding-left: 5px;
}

.modal-footer {
    position: absolute;
    display: flex;
    justify-content: space-around;
    bottom: 0;
}

.modal-footer button {
    cursor: pointer;
    width: 150px;
    height: 35px;
    border-radius: 10px;
    margin: 0 20px 0 20px;
    border: none;
    margin-bottom: 10px;
}

.modal-footer button:hover {
    background-color: rgba(103, 64, 255, 255);
    color: white;
}

@media screen and (max-width:800px)
{
    .modal {
        /* background-color: rgba(0, 0, 0, 0.5); */
        display: flex;
        /* justify-content: center; */
        /* align-items: center; */
        position: fixed;
        margin-left:25%
    }
    .modal-content {
        margin-left: 25%;
        height: 100%;
        border-radius: 20px;
        padding: 30px;
        width: 450px;
        margin-top: -400px;
        background-color: white;
    }

.modal-footer button {
    cursor: pointer;
    width: 120px;
    height: 35px;
    border-radius: 10px;
    margin: 0 20px 0 20px;
    border: none;
    margin-bottom: 10px;
}
}