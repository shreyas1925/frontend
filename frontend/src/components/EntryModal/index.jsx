import React from 'react';
import './EntryModal.css';
import PropTypes from 'prop-types';
export default function EntryModal(props) {
    if (!props.show) {
        return null;
    }
    return (
        <div className="modal" onClick={props.onClose}>
            <div className="modal-contents" onClick={e => e.stopPropagation()}>
                <div className='modal-header'>
                    <span className='modal-title'>New Company Profile</span>
                </div>
                <div className='modal-body'>
                    <div>
                        <span>Name</span>
                        <input type="text" />
                    </div>
                    <div>
                        <span>website</span>
                        <input type="text" />
                    </div>
                    <div>
                        <span>contact</span>
                        <input type="text" />
                    </div>
                    <div>
                        <span>Input</span>
                        <input type="text" />
                    </div>
                    <div>
                        <span>Input</span>
                        <input type="text" />
                    </div>
                    <div>
                        <span>Input</span>
                        <input type="text" />
                    </div>
                </div>
                <div className='modal-footer'>
                    <button onClick={props.onClose} className='modal-close-button'>Close</button>
                    <button className='modal-Add-button'>Add</button>
                </div>
            </div>
        </div>
    );
}

EntryModal.propTypes = {
    onClose: PropTypes.func,
    show: PropTypes.bool,
};