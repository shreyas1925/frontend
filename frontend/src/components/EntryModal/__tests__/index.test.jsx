import EntryModal from "..";
import { render, screen } from "@testing-library/react";

describe("EntryModal", () => {
  it("should render correctly", () => {
    const { asFragment } = render(
      <EntryModal
        onClose={() => {}}
        show={true}
        fields={["name", "description"]}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render correctly with props", () => {
    const { asFragment } = render(
      <EntryModal
        onClose={() => {}}
        show={true}
        fields={["name", "description"]}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
