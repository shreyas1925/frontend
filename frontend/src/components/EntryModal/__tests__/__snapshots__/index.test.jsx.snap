// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntryModal should render correctly 1`] = `
<DocumentFragment>
  <div
    class="modal"
  >
    <div
      class="modal-content"
    >
      <div
        class="modal-header"
      >
        <span
          class="modal-title"
        >
          New Company Profile
        </span>
      </div>
      <div
        class="modal-body"
      >
        <div>
          <span>
            name
          </span>
          <input
            value=""
          />
        </div>
        <div>
          <span>
            description
          </span>
          <input
            value=""
          />
        </div>
      </div>
      <div
        class="modal-footer"
      >
        <button
          class="modal-close-button"
        >
          Close
        </button>
        <button
          class="modal-Add-button"
        >
          Add
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`EntryModal should render correctly with props 1`] = `
<DocumentFragment>
  <div
    class="modal"
  >
    <div
      class="modal-content"
    >
      <div
        class="modal-header"
      >
        <span
          class="modal-title"
        >
          New Company Profile
        </span>
      </div>
      <div
        class="modal-body"
      >
        <div>
          <span>
            name
          </span>
          <input
            value=""
          />
        </div>
        <div>
          <span>
            description
          </span>
          <input
            value=""
          />
        </div>
      </div>
      <div
        class="modal-footer"
      >
        <button
          class="modal-close-button"
        >
          Close
        </button>
        <button
          class="modal-Add-button"
        >
          Add
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
